packages/wattles/lib/src/data_source.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:collection';
+
+import 'package:wattles/wattles.dart';
+
+/// {@template data_source}
+/// The data source is the main entry point for interacting with any database
+/// using the Wattle(s) ORM.
+/// {@endtemplate}
+class DataSource {
+  /// {@macro data_source}
+  DataSource.initialize({
+    required List<Schema> schemas,
+    required this.driver,
+  }) : _schemas = schemas;
+
+  /// The driver that is used to interact with the database.
+  final DatabaseDriver driver;
+
+  /// The schemas that are available in this data source.
+  List<Schema> get schemas => UnmodifiableListView(_schemas);
+  final List<Schema> _schemas;
+
+  /// Returns a [DataStore] instance for the given [Struct].
+  DataStore<T> getStore<T extends Struct>() {
+    final schemas = _schemas.whereType<T>();
```

Comment:
Wondering if there's a use case for subclassing `T` which would break this.

Reply:
That is a valid concern, if anything I would say it shouldn't be allowed. As it would make it more complex for the creation of stores.

Reply:
Right now, I am gonna park the subclassing concern. It will complicate the internals without a massive gain (that I can see).